dashboard/lib/client_state.rb

```diff
@@ -127,6 +141,10 @@ def progress_hash
 
   # Migrates session state to unencrypted cookies.
   def migrate_cookies
+    if session[:puzzle_rating]
+      cookies.permanent[:puzzle_rating] = JSON.generate(session[:puzzle_rating])
```

Comment:
Is there really a session[:puzzle_rating] already?  I didn't spot one in a quick search.